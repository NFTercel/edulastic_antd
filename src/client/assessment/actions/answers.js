import { SET_ANSWER } from "../constants/actions";

export const setUserAnswerAction = (questionId, data) => ({
  type: SET_ANSWER,
  payload: {
    id: questionId,
    data
  }
});
