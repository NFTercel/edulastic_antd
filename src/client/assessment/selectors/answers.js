import { createSelector } from "reselect";

export const getAnswersListSelector = state => state.answers;

export const getAnswerByQuestionIdSelector = questionId =>
  createSelector(
    getAnswersListSelector,
    state => (questionId ? state[questionId] : undefined)
  );
