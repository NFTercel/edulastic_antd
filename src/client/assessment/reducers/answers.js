import { SET_ANSWER, LOAD_ANSWERS, REMOVE_ANSWERS } from "../constants/actions";

const initialState = {};

export default function reducer(state = initialState, { type, payload }) {
  switch (type) {
    case SET_ANSWER:
      return { ...state, [payload.id]: payload.data };
    case LOAD_ANSWERS:
      return {
        ...state,
        ...payload
      };
    case REMOVE_ANSWERS:
      return {};
    default:
      return state;
  }
}
