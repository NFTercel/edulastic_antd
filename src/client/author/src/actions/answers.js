import { CLEAR_ANSWERS } from "../constants/actions";

export const clearAnswersAction = () => ({
  type: CLEAR_ANSWERS
});
