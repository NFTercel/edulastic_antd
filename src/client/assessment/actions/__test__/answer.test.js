import { setUserAnswerAction } from "../answers";

describe("Answer", () => {
  it("set user answer should return an action", () => {
    expect(
      setUserAnswerAction({
        id: "1qjfl125991weplz24667",
        data: {}
      })
    ).toMatchSnapshot();
  });
});
