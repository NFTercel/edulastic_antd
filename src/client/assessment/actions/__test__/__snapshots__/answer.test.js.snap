// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer set user answer should return an action 1`] = `
Object {
  "payload": Object {
    "data": undefined,
    "id": Object {
      "data": Object {},
      "id": "1qjfl125991weplz24667",
    },
  },
  "type": "[answers] set answer",
}
`;
